instrumentation/opentelemetry-instrumentation-logging/src/opentelemetry/instrumentation/logging/__init__.py

```diff
@@ -0,0 +1,201 @@
+will result in KeyError exceptions. Such exceptions are automatically swallowed by the logging module and do not result in crashes but you may still lose out
+on important log messages.
+"""
+# pylint: disable=no-value-for-parameter,no-member,no-name-in-module
+
+import logging  # pylint: disable=import-self
+from os import environ
+
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.logging.environment_variables import (
+    OTEL_PYTHON_LOG_CORRELATION,
+    OTEL_PYTHON_LOG_FORMAT,
+    OTEL_PYTHON_LOG_LEVEL,
+)
+from opentelemetry.trace import (
+    INVALID_SPAN,
+    INVALID_SPAN_CONTEXT,
+    get_current_span,
+    get_tracer_provider,
+)
+
+DEFAULT_LOGGING_FORMAT = "%(asctime)s %(levelname)s [%(name)s] [%(filename)s:%(lineno)d] [trace_id=%(otelTraceID)s span_id=%(otelSpanID)s service_name=%(otelServiceName)s] - %(message)s"
+
+LEVELS = {
+    "debug": logging.DEBUG,
+    "info": logging.INFO,
+    "warning": logging.WARNING,
+    "error": logging.ERROR,
+}
+
+
+class LoggingInstrumentor(BaseInstrumentor):
+    """An instrumentor for stdlib logging module.
+
+    This opt-in instrumentor injects tracing context into logging records and sets the global logging format to the following:
+
+    .. code-block::
+
+        %(asctime)s %(levelname)s [%(name)s] [%(filename)s:%(lineno)d] [signalfx.trace_id=%(sfxTraceId)s signalfx.span_id=%(sfxSpanId)s] - %(message)s
+
+    See `BaseInstrumentor`
+    """
+
+    factory_registered = False
+    is_instrumented = False
+
+    def _instrument(self, **kwargs):
+        LoggingInstrumentor.is_instrumented = True
+        service_name = ""
+        provider = kwargs.get("tracer_provider", None) or get_tracer_provider()
+        resource = provider.resource if provider else None
+        if resource:
+            service_name = resource.attributes.get("service.name")
+
+        old_factory = logging.getLogRecordFactory()
+        if not LoggingInstrumentor.factory_registered:
+            LoggingInstrumentor.factory_registered = True
+
+            def record_factory(*args, **kwargs):
+                record = old_factory(*args, **kwargs)
+                if not LoggingInstrumentor.is_instrumented:
+                    return record
+
+                record.otelSpanID = ""
+                record.otelTraceID = ""
+                record.otelServiceName = service_name
+
+                span = get_current_span()
+                if span != INVALID_SPAN:
+                    ctx = span.get_span_context()
+                    if ctx != INVALID_SPAN_CONTEXT:
+                        record.otelSpanID = format(ctx.span_id, "016x")
+                        record.otelTraceID = format(ctx.trace_id, "032x")
+                return record
+
+            logging.setLogRecordFactory(record_factory)
+
+        if environ.get(OTEL_PYTHON_LOG_CORRELATION, "false").lower() == "true":
```

Comment:
I would argue for the following:

1. remove these environment variables and the basicConfig porcelain altogether. 

If 1 is not acceptable I would argue factoring this out into a separate method, since this is complecting two different behaviors (attaching custom fields to the log record, modifying logging) without a way to use them separately.

Porcelain functions like these typically help the user by orchestrating a complex process that they may not be completely familiar with. In this case, I could achieve the same result as these environment variables by a single call to logging.basicConfig, and I believe it would be pretty common for a python opentelemetry user to be familiar with the logging API.

Reply:
Sure, we can split them into multiple methods or we can add an argument to the `instrument()` function that controls whether to call basicConfig or not in addition to the env var. I think the later would be preferable as all other instrumentations only export `instrument()` and `uninstrument()` functions so adding a 3rd method to this one would need a lot of justification. 

So how about something like `instrument(setup_logging_format=False)`. This would only call basicConfig if the user explicitly passed `setup_logging_fromat=True`. 

WDYT?

Reply:
Added arguments to the `instrument` method to allow users to configure the instrumentation manually through code while keeping env var support to allow `opentelemetry-instrument` command users to configure without having to add any custom code.